server/chat-plugins/helptickets.js

```diff
@@ -758,6 +848,165 @@ const pages = {
+			const nextDate = new Date((prevDate.getMonth() >= 10 ? prevDate.getFullYear() + 1 : prevDate.getFullYear()), (prevDate.getMonth() + 2 > 11 ? (prevDate.getMonth() + 2) - 12 : prevDate.getMonth() + 2));
+			let prevString = Chat.toTimestamp(prevDate).split(' ')[0].split('-', 2).join('-');
+			let nextString = Chat.toTimestamp(nextDate).split(' ')[0].split('-', 2).join('-');
+
+			let buttonBar = '';
+			if (FS(`logs/tickets/${prevString}.csv`).readIfExistsSync()) {
+				buttonBar += `<a class="button" href="/view-help-stats-${table}-${prevString}" target="replace" style="float: left">&lt; Previous Month</a>`;
+			} else {
+				buttonBar += `<a class="button disabled" style="float: left">&lt; Previous Month</a>`;
+			}
+			buttonBar += `<a class="button${table === 'tickets' ? ' disabled"' : `" href="/view-help-stats-tickets-${monthString}" target="replace"`}>Ticket Stats</a> <a class="button ${table === 'staff' ? ' disabled"' : `" href="/view-help-stats-staff-${monthString}" target="replace"`}>Staff Stats</a>`;
+			if (FS(`logs/tickets/${nextString}.csv`).readIfExistsSync()) {
+				buttonBar += `<a class="button" href="/view-help-stats-${table}-${nextString}" target="replace" style="float: right">Next Month &gt;</a>`;
+			} else {
+				buttonBar += `<a class="button disabled" style="float: right">Next Month &gt;</a>`;
+			}
+			let col = query.shift() || '';
+
+			let buf = `<div class="pad ladder"><div style="text-align: center">${buttonBar}</div><br />`;
+			buf += `<table style="margin-left: auto; margin-right: auto"><tbody><tr><th colspan="${table === 'tickets' ? 7 : 3}"><h2 style="margin: 5px auto">Help Ticket Stats - ${date.toLocaleString('en-us', {month: 'long', year: 'numeric'})}</h1></th></tr>`;
+			if (table === 'tickets') {
+				if (!['type', 'totaltickets', 'total', 'initwait', 'wait', 'resolution', 'result'].includes(col)) col = 'type';
+				buf += `<tr><th><Button>type</Button></th><th><Button>totaltickets</Button></th><th><Button>total</Button></th><th><Button>initwait</Button></th><th><Button>wait</Button></th><th><Button>resolution</Button></th><th><Button>result</Button></th></tr>`;
+			} else {
+				if (!['staff', 'num', 'time'].includes(col)) col = 'num';
+				buf += `<tr><th><Button>staff</Button></th><th><Button>num</Button></th><th><Button>time</Button></th></tr>`;
+			}
+
+			/** @type {{[key: string]: (string | number)}[]} */
+			let ticketStats = rawTicketStats.split('\n').map(line => {
+				let splitLine = line.split(',');
+				return {
+					type: splitLine[0],
+					total: parseInt(splitLine[1]),
+					initwait: parseInt(splitLine[2]),
+					wait: parseInt(splitLine[3]),
+					resolution: splitLine[4],
+					result: splitLine[5],
+					staff: splitLine[6],
+				};
+			});
+			if (table === 'tickets') {
+				// {total: num, initwait: num, wait: num, resolution: {dead: num, unresolved: num, resolved: num}, result: num positive, totaltickets: num}
+				/** @type {any} */
```

Comment:
huh? You type this as `any` and add a comment explaining its real type? Why not just use the real type in the first place?

Reply:
Mostly because its a huge pain to get this typed properly, I think I just decided to stop for a bit and wanted to come back to it later but forgot. I'll give it another shot.